components/applications-service/pkg/storage/postgres/hab_event.go

```diff
@@ -134,41 +134,167 @@ func (db *Postgres) IngestHealthCheckEventWithoutMetrics(event *habitat.HealthCh
 
 	// @afiune all our backend was designed for the health check to be all
 	// uppercases but habitat is actually sending case sensitive strings
-	newHealth := strings.ToUpper(event.GetResult().String())
+	eventHealth := strings.ToUpper(event.GetResult().String())
 	// TODO @afiune verify health
 
-	// If the service already exists, we just do a simple update
+	// If the service already exists, we just do an update
 	if exist {
-		return db.updateService(
+		return db.updateTables(
 			svc,
 			eventMetadata,
 			svcMetadata,
 			pkgIdent,
-			newHealth,
+			eventHealth,
 		)
 	}
 
 	// But if the service doesn't exist, we will handle it as a new service insertion
-	err1 := db.insertNewService(
+	err = db.insertNewService(
 		eventMetadata,
 		svcMetadata,
 		pkgIdent,
-		strings.ToUpper(event.GetResult().String()),
+		eventHealth,
 	)
 
-	if err1 != nil {
+	if err != nil {
 		// We will retry once if a unique constraint is hit,
 		// in case a new deployment or service group is in the same batch of events.
-		if err1.Error() == uniqueDeploymentError || err1.Error() == uniqueServiceGroupError {
+		if err.Error() == uniqueDeploymentError || err.Error() == uniqueServiceGroupError {
 			return db.insertNewService(
 				eventMetadata,
 				svcMetadata,
 				pkgIdent,
-				strings.ToUpper(event.GetResult().String()),
+				eventHealth,
 			)
 		}
 	}
-	return err1
+	return err
+}
+
+func (db *Postgres) updateTables(
+	svc *service,
+	eventMetadata *habitat.EventMetadata,
+	svcMetadata *habitat.ServiceMetadata,
+	pkgIdent *packageIdent,
+	health string) error {
+
+	//updates := map[string]interface{}{}
+	db.updateService(svc, eventMetadata, svcMetadata, pkgIdent, health)
+
+	deploy, err := db.getDeployment(svc.DeploymentID)
+	if err != nil {
+		return errors.Wrap(err, "unable to update tables")
+	}
+	db.updateDeployment(deploy, eventMetadata)
+
+	sup, err := db.getSupervisor(svc.SupID)
+	if err != nil {
+		return errors.Wrap(err, "unable to update tables")
+	}
+	db.updateSupervisor(sup, eventMetadata)
+
+	sg, err := db.getServiceGroup(svc.GroupID)
+	if err != nil {
+		return errors.Wrap(err, "unable to update tables")
+	}
+	db.updateServiceGroup(sg, svcMetadata)
+
+	return db.triggerDataUpdates(svc, deploy, sup, sg)
+}
+
+func (db *Postgres) triggerDataUpdates(
+	svc *service,
+	deploy *deployment,
+	sup *supervisor,
+	sg *serviceGroup) error {
+
+	// update only if there is something to update
```

Comment:
This comment is a bit confusing. Does it refer to the the checks like `if svc.needUpdate {` we do for each of the update-able things?

Reply:
Updated